### ISSUE TITLE
Serialization with shared scalar learning rate

### ISSUE THREAD

**Issue by lukedeo** [fchollet/keras#589]
An issue with using the shared scalars for lr and momentum -- now the `to_json` and `to_yaml` methods don't work, as they don't know how to handle theano types directly. I've simply been pickling the dict from the `get_config` method, but I have a project where I need to actually sent the network structure as a `PUT` request. Before I draw up a PR, do you have a preferred way of handing this?

**Comment by fchollet** [comment]
Right, we need to fix this. Basically we'll just `get_value` on the optimizer params during serialization.

> but I have a project where I need to actually sent the network structure as a PUT request

A trivial way to by-pass the issue would be to serialize before compilation (then compile after de-serialization). Then you don't have the optimizer in the config.

**Comment by fchollet** [comment]
Fixed; also added a test. https://github.com/fchollet/keras/commit/6cd8d3c37a2fcaf861e49bfebad49f43cfaa7ab3

**Comment by lukedeo** [comment]
Awesome, thanks!
